### PR TITLE
refactor(cli): implement cli demo function

### LINKED ISSUE #34: create fake records
create a fake pure record, then perform the whole process

refactor the cli. comment the current code there, we will refactor that on a later stage. create a cli with click. as the first feature implement "create fake records". the pure record should be loadable from a xml/json file. Which could be piped directly from the pure rest api or is stored locally.

it should be visible how the record is imported from pure

### PR DESCRIPTION
**Pull request by bengaal**
Implement a cli import demo function with click. This PR closes https://github.com/tu-graz-library/invenio-rdm-pure/issues/34.

### CODE REVIEW

File: invenio_rdm_pure/cli.py [hunk 1]
```diff
@@ -1,3 +1,124 @@
...
+
+
+def create_demo_record() -> dict:
+    """Create a demo record dict."""
+    with open(join(dirname(__file__), "data/pure_record_fake.json"), "rb") as fp:
+        demo_record = json.load(fp)
+
+    faker = Faker()
+
+    demo_record["title"]["value"] = str(faker.text())
```
Comment (utnapischtim) [inline invenio_rdm_pure/cli.py]: question: is it necessary to cast it to a string? the return type of faker.text() should be a string anyway.

File: invenio_rdm_pure/cli.py [hunk 2]
```diff
@@ -1,3 +1,124 @@
...
+    random_number_of_days = random.randrange(days)
+    _date = start_date + timedelta(days=random_number_of_days)
+    return _date.strftime("%Y-%m-%d")
+
+
+def create_invenio_record(record: dict) -> None:
+    """Create a record."""
+    converter = Converter()
+    record_marc21 = converter.convert_pure_json_to_marc21_xml(record)
+    if Marc21Record.is_valid_marc21_xml_string(record_marc21):
```
Comment (utnapischtim) [inline invenio_rdm_pure/cli.py]: Suggestion: My personal preference is to a empty line before of the if statement. Coding blocks improve the readability. Also a personal preference is to return early if a error occur. So in the sense of if not Marc21Record.is_valid_marc21_xml_string: click.secho("...") return. The improvement for the readability is that the if block does not have to be indented.

**Review by utnapischtim — DISMISSED** [review]

File: invenio_rdm_pure/cli.py (hunk 2 above)
Comment (bengaal) [reply invenio_rdm_pure/cli.py]: I see what you mean, the changes have been implemented.

File: invenio_rdm_pure/cli.py (hunk 1 above)
Comment (bengaal) [reply invenio_rdm_pure/cli.py]: Should be a string but I haven't found hard evidence and according to the Pylance linter the return type is `Any`, so I wanted to be sure. The examples in the documentation of Faker however, use it without cast, so I removed the cast.

**Review by utnapischtim — APPROVED** [review]

**Pull request merged**